lib/rack_graphql/exception_json_response_middleware.rb

```diff
@@ -0,0 +1,42 @@
+module RackGraphql
+  class ExceptionJSONResponseMiddleware
+    DEFAULT_ERROR_STATUS_CODE = 500
+
+    def initialize(app, app_name:, error_status_code_map:, log_exception_backtrace:)
+      @app = app
+      @app_name = app_name
+      @error_status_code_map = error_status_code_map
+      @log_exception_backtrace = log_exception_backtrace
+    end
+
+    def call(env)
+      @app.call(env)
+    rescue AmbiguousParamError
+      [
+        400,
+        { 'Content-Type' => 'application/json' },
+        [Oj.dump({})]
+      ]
+    rescue StandardError, LoadError, SyntaxError => e
+      [
+        error_status_code_map[e.class] || DEFAULT_ERROR_STATUS_CODE,
+        { 'Content-Type' => 'application/json' },
+        [Oj.dump('errors' => [exception_hash(e)])]
+      ]
+    ensure
+      ActiveRecord::Base.clear_active_connections! if defined?(ActiveRecord::Base)
```

Comment:
it should be kept in the main middleware, as it needs to be always performed - no matter exception middleware is used or not and no matter exception is raised or not.

Reply:
good point !